File: app/models/group.rb
# == Schema Information
#
# Table name: groups
#
#  id             :integer          not null, primary key
#  name           :string
#  street_address :string
#  city           :string
#  state          :string
#  created_at     :datetime         not null
#  updated_at     :datetime         not null
#  zip_code       :integer
#  domain         :string
#

class Group < ActiveRecord::Base
  has_many :users
  has_many :documents

  validates :name, :street_address, :city, :state, :zip_code, :domain, presence: true
  validates :name, :domain, uniqueness: true
  validates :domain, format: { with: /\A@((?:[-a-z0-9]+\.)+[a-z]{2,})\z/i, on: :create }
end

File: app/assets/javascripts/models/group.js
app.groups.model = {
  all: [],
  new: (function(){
    var counter = 0
    var group = function Group(name, street_address, city, state, zip_code, domain){
      this.name = name
      this.street_address = street_address
      this.city = city
      this.state = state
      this.zip_code = zip_code
      this.domain = domain
      var that = this

      function initialize(){
        counter++;
        that.id = counter;
        app.groups.model.all.push(that);
      };
      initialize();
    }
    return group;
  }()),

  findBy: function findBy(attrHash) {
    var key = Object.keys(attrHash)[0]
    var value = attrHash[key]
    return $.grep(app.groups.model.all, function(group){
      return group[key] == value;
    });
  }
}
